backend/src/schema/resolvers/posts.spec.js

```diff
@@ -140,45 +159,88 @@ describe('Post', () => {
       ).resolves.toMatchObject(expected)
     })
 
-    it('by emotions', async () => {
+    describe('by emotions', () => {
+      const postQueryFilteredByEmotions = gql`
+        query Post($filter: _PostFilter) {
+          Post(filter: $filter) {
+            id
+            emotions {
+              emotion
+            }
+          }
+        }
+      `
+
+      it('filters by single emotion', async () => {
+        const expected = {
+          data: {
+            Post: [
+              {
+                id: 'happy-post',
+                emotions: [{ emotion: 'happy' }],
+              },
+            ],
+          },
+        }
+        await user.relateTo(happyPost, 'emoted', { emotion: 'happy' })
+        variables = { ...variables, filter: { emotions_some: { emotion_in: ['happy'] } } }
+        await expect(
+          query({ query: postQueryFilteredByEmotions, variables }),
+        ).resolves.toMatchObject(expected)
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-11T09:32:20Z" title="Wednesday, September 11th 2019, 11:32:20 am +02:00">Sep 11, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -140,45 +159,88 @@ describe('Post', () => {
       ).resolves.toMatchObject(expected)
     })
 
-    it('by emotions', async () => {
+    describe('by emotions', () => {
+      const postQueryFilteredByEmotions = gql`
+        query Post($filter: _PostFilter) {
+          Post(filter: $filter) {
+            id
+            emotions {
+              emotion
+            }
+          }
+        }
+      `
+
+      it('filters by single emotion', async () => {
+        const expected = {
+          data: {
+            Post: [
+              {
+                id: 'happy-post',
+                emotions: [{ emotion: 'happy' }],
+              },
+            ],
+          },
+        }
+        await user.relateTo(happyPost, 'emoted', { emotion: 'happy' })
+        variables = { ...variables, filter: { emotions_some: { emotion_in: ['happy'] } } }
+        await expect(
+          query({ query: postQueryFilteredByEmotions, variables }),
+        ).resolves.toMatchObject(expected)
```


Here you omit the `expect.arrayContaining` :+1:

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-11T10:33:39Z" title="Wednesday, September 11th 2019, 12:33:39 pm +02:00">Sep 11, 2019</time>_

---


yes, we wanted to use this for the `multiple emotions` test as well, but it failed as I stated :point_up: